compiler/rustc_resolve/src/lib.rs

```diff
@@ -2386,7 +2391,40 @@ impl<'a> Resolver<'a> {
                             .next()
                             .map_or(false, |c| c.is_ascii_uppercase())
                         {
-                            (format!("use of undeclared type `{}`", ident), None)
+                            // Add check case for similarly named item in alternative namespace
+                            let mut suggestion = None;
+
+                            if ribs.is_some() {
+                                if let Some(res) = self.resolve_ident_in_lexical_scope(
+                                    ident,
+                                    ValueNS,
+                                    parent_scope,
+                                    None,
+                                    path_span,
+                                    &ribs.unwrap()[ValueNS],
+                                ) {
+                                    let mut match_span: Option<Span> = None;
+                                    match res {
+                                        LexicalScopeBinding::Res(Res::Local(id)) => {
```

Comment:
To help future readers, could you give an example above each arm (i.e. `Res::Local` or `LexicalScopeBinding::Item`) describing what kind of case each one deals with?